tests/drivers/flash_nop_device/testcase.yaml

```diff
@@ -5,9 +5,13 @@
 #
 tests:
   drivers.flash.flash_nop_device:
-    platform_allow: qemu_x86
+    platform_allow: qemu_x86 native_posix
```

Comment:
If you consider native_posix to be enough for the test, for the `drivers.flash.flash_nop_device.posix_erase_value_0x00` scenario, why do you keep the qemu_x86 here?
The test is the same, the difference is in "erase_value" only, so if native_posix is enough for the other scenario, it should be enough for this.

Reply:
If you agree with the above comment, consider also changing the commit description to one would clarify that qemu_x86 has been replaced with native_posix, rather that the former has been added.

Reply:
Okay, so if I remove `qemu_x86` from this `testcase.yaml` file, do you think, that it is also reasonable to take from `drivers/flash/flash_nop_device/flash_nop_device.c` and `tests/drivers/flash_nop_device/src/main.c` this part:

```
/* configuration derived from DT */
#ifdef CONFIG_ARCH_POSIX
#define SOC_NV_FLASH_NODE DT_CHILD(DT_INST(0, zephyr_sim_flash), flash_0)
#else
#define SOC_NV_FLASH_NODE DT_CHILD(DT_INST(0, zephyr_sim_flash), flash_sim_0)
#endif /* CONFIG_ARCH_POSIX */
```

and simplify it into this:

```
/* configuration derived from DT */
#define SOC_NV_FLASH_NODE DT_CHILD(DT_INST(0, zephyr_sim_flash), flash_0)
```

Reply:
Yes, you are right. If we find qemu_x86 test not needed then we neither need code that supports them.